### DISCUSSION TITLE
How do I solve Lesson 4 challenge ?

### DISCUSSION THREAD

**Discussion by 0xgun** [Cyfrin/foundry-full-course-cu#218] [Q&A]
I get this kind of error while submitting the value 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/86967803/88937c92-4a50-45c1-8318-18b8dd53dcff)

**Comment by shikhar229169** [comment]
Hey @0xgun, ensure that you have enough eth to pay for gas and the value that you are passing is correct. Do let me know how you are getting the pricefeeds value and what you are passing to contract function parameters.

**Reply by 0xgun** [reply]
So I deployed this contract 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/86967803/fa12b3be-498a-43e3-a5da-f978c44c5151)

and I'm able to to mint it on sepolia network but on Arbitrum it shows error

**Reply by 0xtdey** [reply]
on Arbitrum it requires real money to mint the NFT

**Reply by 0xgun** [reply]
I have eth on Arbitrum chain  and also I Minted other Lesson's NFT's on Arbitrum

**Reply by rizoom** [reply]
This is probably because price is multiplied by `10e10`, not `1e10`. See my first comment https://github.com/Cyfrin/foundry-full-course-f23/discussions/218#discussioncomment-6341118

So you probably need one additional zero.
At least this is the case for the Arbitrum contract. I haven't checked the Sepolia version.

**Reply by komi786** [reply]
Thank you @rizoom  and @tamoghna-dey . I was struggling with Lesson 4 challenge and your discussion helped me deploy the contract successfully.

**Reply by 0xtdey** [reply]
Welcome :)

**Reply by xKeywordx** [reply]
@rizoom that's the solution, lol. So instead of having 18 decimals, you need to add 19, because the conde is 10e10 instead of 1e10 :)
Thank you for the help

**Comment by 0xtdey** [comment]
Make sure your wallet has enough SepoliaETH to pass the transaction, and if not you can head over to Alchemy sepolia faucet to get SepoliaETH

**Comment by rizoom** [comment]
I had the same error, if the answer is wrong the gas calculation is totally off (metamask was requesting 100k eth for gas on my side!)
In fact the challenge requires you to submit a price with 19 decimals, and not 18 as you might have guessed.

I am actually unsure if this was done on purpose, in the smart contract code the actual price is multiplied by `10e10` (ie. `1e11`), I wonder if it is a trap or a mistake 😉
So be sure to submit a guess price with 19 decimals and it should solve the gas issue as well.

**Reply by 0xtdey** [reply]
Did you solve challenge 4?

**Reply by rizoom** [reply]
Yes 👍

**Reply by 0xtdey** [reply]
The price feed returns price of ETH in USD, and in the challenge, the correct answer can be +-3 ETH, can you give a hint?
What's the answer in is it in USD? I tried it but it didn't work
And if it is ETH, Wei, Gwei how can price of ETH be in ETH?

**Reply by rizoom** [reply]
Yes it is the price of Eth in usd, but the final price to submit is a (big) integer that contains many decimals (1e8 * **10**e10).

**Reply by 0xtdey** [reply]
But the price guess is in uint256 and it isn't taking decimal numbers, so how can we do that?

**Reply by rizoom** [reply]
Yes you submit an integer which contains the decimal part of the price, so what you submit is a very large number which represent the price with decimals.
Ex: If it was 4 decimals, `123.4567` would be `1234567`.
 
If it is still unclear I'd recommend to watch again the part about how to handle decimals in lesson 4.

**Reply by 0xtdey** [reply]
Yaa I got this part and understood it, as of now the price of ETH in USD from price feed as of 4/7/23 is 195379669478, after putting the decimal places it becomes 1953.79669478, which is having 8 digits after the decimal. 

According to the challenge contract it should then be multiplied with 10e10 = 10x10^10 = 10^11
So , 1953.79669478 x 10^11 = 195379669478000, but it still shows wrong answer.

**Reply by rizoom** [reply]
You are close, it is just that you don't need to "put decimal back" in the price feed. You know that the price feed has 8 decimals, but you are still using the plain integer value `195379669478`.

That is why I am talking about 19 decimals in my first post, `1e8` (from price feed value) * `10e10` (from challenge contract).
Hope this helps!

**Reply by 0xtdey** [reply]
Thanks a lot
